lib/stripe.rb

```diff
@@ -303,7 +348,18 @@ def self.api_error(error, resp, error_obj)
     APIError.new(error[:message], resp.code, resp.body, error_obj, resp.headers)
   end
 
-  def self.handle_restclient_error(e, api_base_url=nil)
+  def self.handle_restclient_error(e, request_opts, retry_count, api_base_url=nil)
+
+    if should_retry?(e, retry_count)
+      retry_count = retry_count + 1
+      sleep sleep_time(retry_count)
+      response = execute_request_with_rescues(request_opts, api_base_url, retry_count)
+      if self.on_successful_retry
```

Comment:
I wanted to know if we were ever actually hitting this thing, so I put some logging in that callback.  That's how I know this code has actually caught errors for us.  (A bit of an edge case, for sure.  I wouldn't complain too much if you wanted to remove it)